From 44031a94ea087140677c927a9c04626483a9bc8e Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Tue, 8 Nov 2011 23:20:53 +0800
Subject: [PATCH 170/280] ENGR00161618 vpu: revert the patch of ENGR00161320
 since f/w is updated

Revert "ENGR00161320 vpu: workaround to disable BWB flag on i.mx6q"

This reverts commit 263436931d36368746f661fb10afcf808badbcf2 since
updated v2.1.2 of r25183 SourceCodeRev fixed the issue.
---
 vpu/vpu_lib.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index fc24543..9beb8b9 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -211,7 +211,10 @@ RetCode vpu_Init(void *cb)
 		    BUF_PIC_FLUSH << BIT_BUF_PIC_FLUSH | BUF_PIC_RESET <<
 		    BIT_BUF_PIC_RESET;
 		VpuWriteReg(BIT_BIT_STREAM_CTRL, data);
-		VpuWriteReg(BIT_FRAME_MEM_CTRL, IMAGE_ENDIAN);
+		if (cpu_is_mx6q())
+			VpuWriteReg(BIT_FRAME_MEM_CTRL, IMAGE_ENDIAN | 1 << 12);
+		else
+			VpuWriteReg(BIT_FRAME_MEM_CTRL, IMAGE_ENDIAN);
 		VpuWriteReg(BIT_INT_ENABLE, 1 << INT_BIT_PIC_RUN);
 		VpuWriteReg(BIT_AXI_SRAM_USE, 0);	/* init to not use SRAM */
 
@@ -2467,6 +2470,7 @@ RetCode vpu_DecOpen(DecHandle * pHandle, DecOpenParam * pop)
 
 		if (pDecInfo->mapType)
 			val |= (pDecInfo->tiledLinearEnable << 11 | 0x03 << 9);
+		val |= 1 << 12;
 	}
 	pCodecInst->ctxRegs[CTX_BIT_FRAME_MEM_CTRL] =
 		    val | (pDecInfo->openParam.chromaInterleave << 2);
@@ -3252,7 +3256,7 @@ RetCode vpu_DecStartOneFrame(DecHandle handle, DecParam * param)
 	DecInfo *pDecInfo;
 	DecParam *pDecParam;
 	Uint32 rotMir;
-	Uint32 val = 0, i;
+	Uint32 val = 0;
 	RetCode ret;
 
 	ENTER_FUNC();
-- 
1.8.0

